google/cloud/spanner/result_set.h

```diff
@@ -33,10 +34,182 @@ class ResultSetSource {
   // Returns OK Status with no Value to indicate end-of-stream.
   virtual StatusOr<optional<Value>> NextValue() = 0;
   virtual optional<google::spanner::v1::ResultSetMetadata> Metadata() = 0;
+  virtual std::int64_t RowsModified() const = 0;
+  virtual optional<std::unordered_map<std::string, std::string>> QueryStats()
+      const = 0;
+  virtual optional<QueryPlan> QueryExecutionPlan() const = 0;
+
   virtual optional<google::spanner::v1::ResultSetStats> Stats() = 0;
 };
 }  // namespace internal
 
+/**
+ * Represents the result of a read operation using `spanner::Client::Read()`.
+ *
+ * Note that a `ReadResult` returns both the data for the operation, as a
```

Comment:
"both" doesn't sound right in this sentence.

Reply:
Done.